contracts/controller/src/healthfactor.rs

```diff
@@ -34,13 +35,40 @@ impl Contract {
 
         self.calculate_assets_weighted_price(&map_raw)
     }
+
+    pub fn calculate_accrued_borrow_interest(&self, account_id: AccountId) -> Balance {
+        let mut total_accrued_interest = 0;
+        let user_profile = self.user_profiles.get(&account_id).unwrap_or_default();
+        let total_borrows = user_profile
+            .account_borrows
+            .iter()
+            .map(|(_, balance)| balance)
+            .sum();
+
+        for (token_address, borrow_data) in user_profile.borrow_data.iter() {
+            let accrued_interest = Ratio(total_borrows)
+                * borrow_data.borrow_rate
+                * Ratio((block_height() - borrow_data.borrow_block) as u128)
+                / RATIO_DECIMALS;
+
+            let price = self.get_price(token_address.clone()).unwrap();
+            let accrued_interest_amount = Percentage::from(price.volatility.0).apply_to(
+                Balance::from(price.value) * accrued_interest.0 / 10u128.pow(price.fraction_digits),
+            );
+
+            total_accrued_interest += accrued_interest_amount;
+        }
+        total_accrued_interest
+    }
 }
 
 #[near_bindgen]
 impl Contract {
     pub fn get_health_factor(&self, user_account: AccountId) -> Ratio {
         let collaterals = self.get_account_sum_per_action(user_account.clone(), ActionType::Supply);
-        let borrows = self.get_account_sum_per_action(user_account, ActionType::Borrow);
+        let mut borrows = self.get_account_sum_per_action(user_account.clone(), ActionType::Borrow);
```

Comment:
@dariacherniak in rust it's preferable to avoid mutability you either can shadow borrow with new value
```
let borrows = self.get_account_sum_per_action(user_account.clone(), ActionType::Borrow); 
let borrows = borrows + self.calculate_accrued_borrow_interest(user_account);
```

or calculate it inplace
```
let borrows = self.get_account_sum_per_action(user_account.clone(), ActionType::Borrow) + self.calculate_accrued_borrow_interest(user_account);
```